exonum-java-binding/core/rust/src/runtime/java_runtime_factory.rs

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use jni::{
+    self,
+    errors::{Error, ErrorKind},
+    objects::{GlobalRef, JObject},
+    Executor, InitArgs, InitArgsBuilder, JavaVM, Result as JniResult,
+};
+
+use exonum::runtime::Runtime;
+use runtime::config::{self, Config, InternalConfig, JvmConfig, RuntimeConfig};
+use std::{path::Path, sync::Arc};
+use utils::{convert_to_string, panic_on_exception, unwrap_jni};
+use JavaRuntimeProxy;
+
+const SERVICE_RUNTIME_BOOTSTRAP_PATH: &str = "com/exonum/binding/app/ServiceRuntimeBootstrap";
+const CREATE_RUNTIME_ADAPTER_SIGNATURE: &str =
+    "(L/java/lang/String;I)Lcom/exonum/binding/core/runtime/ServiceRuntimeAdapter;";
+
+/// Creates new runtime from provided config.
+///
+/// There can be only one `JavaServiceRuntime` instance at a time.
```

Comment:
This is not entirely accurate as JavaServiceRuntime does not manage the VM, and
/// there can be multiple runtimes (in native tests, when created by testkit, etc.).
/// I'd clarify the wording, and consider if we need to represent a runtime-managing-VM-and-runtime.